plugins/GCodeWriter/GCodeWriter.py

```diff
@@ -21,6 +22,40 @@ def write(self, stream, node, mode = MeshWriter.OutputMode.TextMode):
         if gcode_list:
             for gcode in gcode_list:
                 stream.write(gcode)
+            profile = self._serialiseProfile(Application.getInstance().getMachineManager().getActiveProfile()) #Serialise the profile and put them at the end of the file.
+            stream.write(profile)
             return True
 
         return False
+
+    ##  Serialises the profile to prepare it for saving in the g-code.
+    #
+    #   The profile are serialised, and special characters (including newline)
+    #   are escaped.
+    #
+    #   \param profile The profile to serialise.
+    #   \return A serialised string of the profile.
+    def _serialiseProfile(self, profile):
+        version = 1 #IF YOU CHANGE THIS FUNCTION IN A WAY THAT BREAKS REVERSE COMPATIBILITY, INCREMENT THIS VERSION NUMBER!
+        prefix = ";SETTING_" + str(version) + " " #The prefix to put before each line.
+        
+        serialised = profile.serialise()
+        
+        #Escape characters that have a special meaning in g-code comments.
+        escape_characters = { #Which special characters (keys) are replaced by what escape character (values).
+                              #Note: The keys are regex strings. Values are not.
+            "\\": "\\\\", #The escape character.
+            "\n": "\\n",  #Newlines. They break off the comment.
+            "\r": "\\r"   #Carriage return. Windows users may need this for visualisation in their editors.
+        }
+        escape_characters = dict((re.escape(key), value) for key, value in escape_characters.items())
+        pattern = re.compile("|".join(escape_characters.keys()))
+        serialised = pattern.sub(lambda m: escape_characters[re.escape(m.group(0))], serialised) #Perform the replacement with a regular expression.
```

Comment:
Right now it does all the replacements with one big regular expression. I looked for a better solution but the only real alternative is to iterate over the elements of the dictionary and explicitly substitute one by one, but that has the disadvantage that the result of the first substitution can be interpreted as an escaped character in the second: `\\n` gets unescaped to `\n` which gets unescaped to a newline character.

Python has built-in methods for escaping and unescaping for python code, URLs and a few others, but not for g-code. There is no built-in method for replacing multiple strings at the same time.